github_actions/lib/dependabot/github_actions/update_checker.rb

```diff
@@ -87,13 +85,39 @@ def fetch_latest_version_for_git_dependency
         nil
       end
 
+      def latest_version_tag
+        @latest_version_tag ||= begin
+          return git_commit_checker.local_tag_for_latest_version if dependency.version.nil?
+
+          latest_tags = git_commit_checker.local_tags_for_latest_version_commit_sha
+
+          # Find the latest version with the same precision as the pinned version.
+          # Falls back to a version with the closest precision if no exact match.
```

Comment:
I wasn't sure what to do here but assumed we'd probably run into this somewhere in the wild. A simpler option would be to refuse to update if we couldn't find a match?

Reply:
I think this approach is good, I think both outcomes are mildly surprising but failing over to _some_ kind of update at least alerts the user to the new version and highlights that the tagging strategy on the library has changed. 

Cases where we don't do anything are harder to rely to the user without them reading the logs closely, so I think this is the better outcome.

Reply:
Do we have a test that covers the tie break behaviour? 

I haven't gone over the existing tests outside the diff, but it occurs to me we might already be testing this as a variation on the current behaviour but I couldn't spot it.

Reply:
Nope, there were not. It's a bit of work to setup test fixtures for each case so I wanted to be sure we were OK with the approach first. I added some tests in https://github.com/dependabot/dependabot-core/pull/4953/commits/aaa77031e5ae9a33bf23f3d4b7a6c53198d19383 by stubbing out the call to GitCommitChecker so the examples are easier to read.